solidity_bridge/src/Treasury.sol

```diff
@@ -0,0 +1,79 @@
+pragma solidity ^0.8.24;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+import {ITreasury, WithdrawMessage} from "./interfaces/ITreasury.sol";
+import {Constants} from "./libraries/Constants.sol";
+import {IMessageQueue, IMessageQueueReceiver, VaraMessage} from "./interfaces/IMessageQueue.sol";
+
+contract Treasury is ITreasury, Context, AccessControl, IMessageQueueReceiver {
+    using SafeERC20 for IERC20;
+
+    bytes32 private constant GRC_20_GATEWAY_ADDRESS =
+        bytes32(
+            0x0707070707070707070707070707070707070707070707070707070707070707
```

Comment:
isn't that a mock address?

Reply:
That is the address in Vara blockchain, I suggest dong the same for Ethereal, setting address of MessageQueue installed of using Roles  from OpenZeppelin

Reply:
> setting address of MessageQueue instead of using Roles from OpenZeppelin

Well,  I agree, that could make things easier

But this gateway mock address - can't be merged, I believe. Better remove `constant` and just set it during initialization as a state variable.

Reply:
That will not work well because in that case we have to be able to change this memory data, and that means we need to create ADMIN role. I strongly recommend use constant values in our case.